pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java

```diff
@@ -57,7 +57,7 @@ public String getServerVersion() {
     public int getServerVersionNum() {
         if (serverVersionNum != 0)
             return serverVersionNum;
-        return Utils.parseServerVersionStr(serverVersion);
+        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
```

Comment:
There are a couple of places where internal threads touch it, but we
control those so we should be able to eliminate much of the synchronization

Dave Cramer

On 27 December 2015 at 16:39, Valentine Gogichashvili <
notifications@github.com> wrote:

> In pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java
> <https://github.com/pgjdbc/pgjdbc/pull/464#discussion_r48458515>:
>
> > @@ -57,7 +57,7 @@ public String getServerVersion() {
> >      public int getServerVersionNum() {
> >          if (serverVersionNum != 0)
> >              return serverVersionNum;
> > -        return Utils.parseServerVersionStr(serverVersion);
> > +        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
>
> I wonder if the synchronization is needed at all here. The point that
> @zapov <https://github.com/zapov> raised about no real reason to
> synchronize for the getServerVersionNum should apply to most of the methods
> in this class.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/464/files#r48458515>.
>